src/web/mjs/engine/Connector.mjs

```diff
@@ -476,6 +476,28 @@ export default class Connector {
             } );
     }
 
+    fetchGraphQL(request, operationName, query, variables) {
+        if( typeof request === 'string' ) {
+            request = new Request( request, this.requestOptions );
+        }
+        // TODO: check if this will affect (replace) the input parameter?
+        if( request instanceof URL ) {
+            request = new Request( request.href, this.requestOptions );
+        }
+
+        const graphQLRequest = new Request(request, {
```

Comment:
Create and assign new request options, to ensure only certain options should be overwritten
```javascript
const graphQLRequest = new Request(request, Object.assign(this.requestOptions, {
            method: 'POST',
            body: JSON.stringify({
                operationName,
                query,
                variables,
            })
        }));
```